### PR TITLE
AST-42345 inner tool to install an executable from a zip file (SCAResolver and Vorpal tools)

### PR DESCRIPTION
**Pull request by tamarleviCm**
### Description

> Extract the ScaResolver installation to a general place
the installer check if new version is available,
download the installable file
create a work dir for the executable 
and unzip the file there.

adding a configuration for 'vorpal' installation for each OS

### References

>  https://checkmarx.atlassian.net/browse/AST-42345

### Testing

> Manual tests for the vorpal installation - download and unzip for each os
> Unit test for the installation logic
> Unit test for Linux vorpal installation (our tests run on Linux, but we can run the test on each os)

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/sastrealtime/sast_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	"os"
+	"testing"
+
+	"github.com/checkmarx/ast-cli/internal/commands/sastrealtime/sastconfiguration"
+	"github.com/checkmarx/ast-cli/internal/commands/scarealtime/scaconfiguration"
+	"github.com/checkmarx/ast-cli/internal/services/osinstaller"
+	"gotest.tools/assert"
+)
+
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/sastrealtime/sast_test.go:12]: For tests, please follow the convention:
UnitOfWork_StateUnderTest_ExpectedBehavior

UnitOfWork: This is the method or function being tested.
StateUnderTest: This describes the conditions under which the test is run.
ExpectedBehavior: This describes the expected outcome of the test.

Look at some references in project_test.go, first tests in result_test.go and some other files

File: internal/commands/sastrealtime/sastconfiguration/sast-linux.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//go:build linux
+
+package sastconfiguration
+
+import (
+	"github.com/checkmarx/ast-cli/internal/services/osinstaller"
+)
+
+var Params = osinstaller.InstallationConfiguration{
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/sastrealtime/sastconfiguration/sast-linux.go:9]: The values in the configuration are duplicate in the sast-mac, sast-linux and sast-windows...
it's better to have them as constants and if and when they change - you'd only have to change them once.
Notice we have a constants folder, you can add a new file there.

File: internal/commands/scarealtime/sca-realtime-utils.go [hunk 3]
```diff
@@ -38,156 +20,3 @@ var GetPackageManagerFromResolvingModuleType = map[string]string{
 	"nuget":     "Nuget",
 	"cpp":       "Cpp",
 }
-
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/scarealtime/sca-realtime-utils.go:41]: This code removal is related to the feature?

File: internal/commands/scarealtime/scaconfiguration/sca-linux.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+//go:build linux
```
Comment (checkmarx-kobi-hagmi) [inline internal/commands/scarealtime/scaconfiguration/sca-linux.go:1]: Also here - when possible - better to extract the string values to constants.

**Review by OrShamirCM — COMMENTED** [review]
Looks good!
Please have camel case in the folder names

**Review by tamarleviCm — COMMENTED** [review]
https://go.dev/blog/package-names

it seem like the convention of package name in golang is only lower case 
but it has to be short so I will short the packages names

File: test/integration/scan_test.go [hunk 5]
```diff
@@ -928,7 +929,8 @@ func copyResultsToTempDir() error {
 		return err
 	}
 	// Write data to dst
-	err = ioutil.WriteFile(realtime.ScaResolverResultsFileNameDir, data, 0644)
+	scaResolverResultsFileNameDir := filepath.Join(scaconfig.Params.WorkingDir(), realtime.ScaResolverResultsFileName)
+	err = ioutil.WriteFile(scaResolverResultsFileNameDir, data, 0644)
```
Comment (AlvoBen) [inline test/integration/scan_test.go:933]: suggestion: ioutil is deprecated, consider changing it to os.WriteFile
Note: There are several places in this pr that use ioutil

**Review by AlvoBen — APPROVED** [review]

**Pull request merged**